core/new-gui/src/app/workspace/service/execute-workflow/execute-workflow.service.ts

```diff
@@ -396,7 +405,11 @@ export class ExecuteWorkflowService {
       );
     };
 
-    const operators: LogicalOperator[] = workflowGraph.getAllEnabledOperators().map(op => {
+    const subDag = isDefined(targetOperatorId) ? workflowGraph.getSubDag(targetOperatorId) : undefined;
+
+    const operators: LogicalOperator[] = (
+      isDefined(subDag) ? subDag.operators : workflowGraph.getAllEnabledOperators()
```

Comment:
can we restructure this part to be the following for better readability?
```
if(isDefined(subDag)){
    //generate request based on subDag
} else{
    // the original logic
}
```

Reply:
I revised the code for readability. please check.